docs/machine-learning/how-to-guides/prepare-data-ml-net.md

```diff
@@ -0,0 +1,331 @@
+
+## Use normalizers
+
+[Normalization](https://en.wikipedia.org/wiki/Feature_scaling) is a data pre-processing technique used to standardize features that are not on the same scale. For example, the ranges for values like age and income vary significantly with age generally being in the range of 0-100 and income generally being in the range of zero to thousands. 
+
+ML.NET provides the following normalizer options:
+
+- [Binning](xref:Microsoft.ML.NormalizationCatalog.NormalizeBinning*)
+- [GlobalContrast](xref:Microsoft.ML.NormalizationCatalog.NormalizeGlobalContrast*)
+- [LogMeanVariance](xref:Microsoft.ML.NormalizationCatalog.NormalizeLogMeanVariance*)
+- [LpNorm](xref:Microsoft.ML.NormalizationCatalog.NormalizeLpNorm*)
+- [MeanVariance](xref:Microsoft.ML.NormalizationCatalog.NormalizeMeanVariance*)
+- [MinMax](xref:Microsoft.ML.NormalizationCatalog.NormalizeMinMax*)
+- [SupervisedBinning](xref:Microsoft.ML.NormalizationCatalog.NormalizeSupervisedBinning*)
+
+### Min-Max normalization
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+HomeData[] homeDataList = new HomeData[]
+{
+    new HomeData
+    {
+        NumberOfBedrooms = 2f,
+        Price = 200000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms = 1f,
+        Price = 100000f
+    }
+};
+```
+
+Normalize the data using min-max normalization using the [`NormalizeMinMax`](xref:Microsoft.ML.NormalizationCatalog.NormalizeMinMax*) method.
+
+```csharp
+// Define min-max estimator
+var minMaxEstimator = mlContext.Transforms.NormalizeMinMax("Price");
+
+// Build ITransformer that performs operation
+ITransformer minMaxTransformer = minMaxEstimator.Fit(data);
+
+// Transform data
+IDataView transformedData = minMaxTransformer.Transform(data);
+```
+
+The original price values `[200000,100000]` are converted to `[ 1, 0.5 ]` using the `MinMax` normalization formula.
+
+### Binning
+
+[Binning](https://en.wikipedia.org/wiki/Data_binning) converts continuous values into a discrete representation. For example, suppose one of your features is age. Instead of using the actual age value, with binning you can create ranges for that value. 0-18 could be one bin, another could be 19-35 and so on. 
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+HomeData[] homeDataList = new HomeData[]
+{
+    new HomeData
+    {
+        NumberOfBedrooms=1f,
+        Price=100000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms=2f,
+        Price=300000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms=6f,
+        Price=600000f
+    }
+};
+```
+
+Normalize the data into bins using the [`NormalizeBinning`](xref:Microsoft.ML.NormalizationCatalog.NormalizeBinning*) method. The `maximumBinCount` parameter enables you to specify how many bins you want to put your data into. In this example, data will be put into two bins.  
```

Comment:
"specify how many bins you want to put your data into" -> "specify the number of bins needed to classify your data."